scripts/build-doc-list.js

```diff
@@ -7,20 +8,34 @@ const { slugify } = require('markdown-toc/lib/utils')
 const readingTime = require('reading-time')
 const { markdownToTxt } = require('markdown-to-txt')
 
-let specWeight = 100
-const result = []
+const yargs = require('yargs/yargs')
+const { hideBin } = require('yargs/helpers')
+const argv = yargs(hideBin(process.argv)).argv
+
 const basePath = 'pages'
-const postDirectories = [
+const defaultDirectoriesToProcess = [
   [`${basePath}/docs`, '/docs'],
   [`${basePath}/blog`, '/blog'],
   [`${basePath}/about`, '/about'],
-  [`${basePath}/jobs`, '/jobs'],
-]
-walkDirectories(postDirectories, result)
+  [`${basePath}/jobs`, '/jobs']
+];
+
+let specWeight = 100
+const result = []
+
+const output = argv.output;
+const groupCategoriesByFrontMatter = !!argv.groupByCategories;
+const directories = !argv.directory ? defaultDirectoriesToProcess : [[argv.directory]];
```

Comment:
```suggestion
const directories = argv.directory ? [[argv.directory]] : defaultDirectoriesToProcess;
```

I know that it's the same but it's easier to read :)

Reply:
Thanks @magicmatatjahu , I have removed this now, and refactored things slightly, hopefully for the best... :O